src/utils.rs

```diff
@@ -5,9 +5,13 @@ use gdal_sys::CPLErr;
 
 use crate::errors::*;
 
-pub fn _string(raw_ptr: *const c_char) -> String {
-    let c_str = unsafe { CStr::from_ptr(raw_ptr) };
-    c_str.to_string_lossy().into_owned()
+pub fn _string(raw_ptr: *const c_char) -> Option<String> {
```

Comment:
It might be worth to add a documentation comment to this function and to `__pathbuf` to point out that they perform null checks internally.